templates/criterion.js

```diff
@@ -0,0 +1,870 @@
+                finalLow, -1.0);
+            var h = easing === 1 ? finalHigh :
+              conservativeError(b._view.x, b._model.x,
+                high, finalHigh, 1.0);
+            drawErrorBar(chart, ctx, l, h, b._view.y, 4, errorBar.color);
+          });
+        });
+      },
+    };
+  })();
+
+  // Formats the ticks on the X-axis on the scatter plot
+  var iterFormatter = function() {
+    var denom = 0;
+    return function(iters, index, values) {
+      if (iters == 0) {
+        return '';
+      }
+      if (index == values.length - 1) {
+        return '';
+      }
+      var power;
+      if (iters >= 1e9) {
+        denom = 1e9;
+        power = '⁹';
+      } else if (iters >= 1e6) {
+        denom = 1e6;
+        power = '⁶';
+      } else if (iters >= 1e3) {
+        denom = 1e3;
+        power = '³';
+      } else {
+        denom = 1;
+      }
+      if (denom > 1) {
+        var value = (iters / denom).toFixed();
+        return String(value) + '×10' + power;
+      } else {
+        return String(iters);
+      }
+    };
+  };
+
+  var colors = ["#edc240", "#afd8f8", "#cb4b4b", "#4da74d", "#9440ed"];
+  var errorColors = ["#cda220", "#8fb8d8", "#ab2b2b", "#2d872d", "#7420cd"];
+
+
+  // Positions tooltips at cursor. Required for overview since the bars may
+  // extend past the canvas width.
+  Chart.Tooltip.positioners.cursor = function(_elems, position) {
+    return position;
+  }
+
+  function axisType(logaxis) {
+    return logaxis ? 'logarithmic' : 'linear';
+  }
+
+  function reportSort(a, b) {
+    return a.reportNumber - b.reportNumber;
+  }
+
+  // adds groupNumber and group fields to reports;
+  // returns list of list of reports, grouped by group
+  function groupReports(reports) {
+
+    function reportGroup(report) {
+      var parts = report.groups.slice();
+      parts.pop();
+      return parts.join('/');
+    }
+
+    var groups = [];
+    reports.forEach(function(report) {
+      report.group = reportGroup(report);
+      if (groups.length === 0) {
+        groups.push([report]);
+      } else {
+        var prevGroup = groups[groups.length - 1];
+        var prevGroupName = prevGroup[0].group;
+        if (prevGroupName === report.group) {
+          prevGroup.push(report);
+        } else {
+          groups.push([report]);
+        }
+      }
+      report.groupNumber = groups.length - 1;
+    });
+    return groups;
+  }
+
+  // compares 2 arrays lexicographically
+  function lex(aParts, bParts) {
+    for(var i = 0; i < aParts.length && i < bParts.length; i++) {
+      var x = aParts[i];
+      var y = bParts[i];
+      if (x < y) {
+        return -1;
+      }
+      if (y < x) {
+        return 1;
+      }
+    }
+    return aParts.length - bParts.length;
+  }
+  function lexicalSort(a, b) {
+    return lex(a.groups, b.groups);
+  }
+
+  function reverseLexicalSort(a, b) {
+    return lex(a.groups.slice().reverse(), b.groups.slice().reverse());
+  }
```

Comment:
This is the implementation for the `lexical` and `colexical` sort orders. The implementation is the same, except that the colexical first reverses the two group arrays performing a normal `lexical` sort.